src/Common/src/CoreLib/System/Resources/ResourceReader.core.cs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Resources
+{
+    partial class ResourceReader
+    {
+        private readonly bool _permitDeserialization;  // can deserialize BinaryFormatted resources
+        private object? _binaryFormatter; // binary formatter instance to use for deserializing
+
+        // statics used to dynamically call into BinaryFormatter
+        // When successfully located s_binaryFormatterType will point to the BinaryFormatter type
+        // and s_deserializeMethod will point to an unbound delegate to the deserialize method.
+        private static Type s_binaryFormatterType;
+        private static Func<object?, Stream, object> s_deserializeMethod;
+
+        // This is the constructor the RuntimeResourceSet calls,
+        // passing in the stream to read from and the RuntimeResourceSet's 
+        // internal hash table (hash table of names with file offsets
+        // and values, coupled to this ResourceReader).
+        internal ResourceReader(Stream stream, Dictionary<string, ResourceLocator> resCache, bool permitDeserialization)
+        {
+            Debug.Assert(stream != null, "Need a stream!");
+            Debug.Assert(stream.CanRead, "Stream should be readable!");
+            Debug.Assert(resCache != null, "Need a Dictionary!");
+
+            _resCache = resCache;
+            _store = new BinaryReader(stream, Encoding.UTF8);
+
+            _ums = stream as UnmanagedMemoryStream;
```

Comment:
>UnmanagedMemoryStream [](start = 29, length = 21)

would it be possible (in the future) we pass different types of streams? or this is unlikely?

Reply:
This is existing code that was moved.  It does support different streams, in those case _ums is null and it will use it as a stream.